coderd/reaper/detector.go

```diff
@@ -210,11 +230,11 @@ func (d *Detector) run(t time.Time) Stats {
 	return stats
 }
 
-func unhangJob(ctx context.Context, log slog.Logger, db database.Store, pub pubsub.Pubsub, jobID uuid.UUID) error {
+func reapJob(ctx context.Context, log slog.Logger, db database.Store, pub pubsub.Pubsub, jobID uuid.UUID) error {
 	var lowestLogID int64
 
 	err := db.InTx(func(db database.Store) error {
-		locked, err := db.TryAcquireLock(ctx, database.GenLockID(fmt.Sprintf("hang-detector:%s", jobID)))
+		locked, err := db.TryAcquireLock(ctx, database.GenLockID(fmt.Sprintf("reaper:%s", jobID)))
 		if err != nil {
 			return xerrors.Errorf("acquire lock: %w", err)
 		}
```

Comment:
Should be able to drop this entirely if you use the `SELECT ... FOR UPDATE SKIP LOCKED` query